tf2/include/tf2/time.h

```diff
@@ -52,7 +53,24 @@ namespace tf2
     return std::chrono::system_clock::now();
   }
 
-  inline double durationToSec(const tf2::Duration & input){
+  inline Duration durationFromSec(double t_sec)
+  {
+    uint32_t sec, nsec;
+    sec = (uint32_t)floor(t_sec);
+    nsec = (uint32_t)std::round((t_sec-sec) * 1e9);
```

Comment:
@wjwwood thanks for the review -- this PR incorporates commits from another, so I've moved your comment to the underlying PR https://github.com/ros2/geometry2/pull/16/commits/2a1bce74429eec3e60157f28f068a41da053754b#r114168134